warplib/src/test/warp_memory_test.cairo

```diff
@@ -0,0 +1,307 @@
+use warplib::warp_memory::WarpMemory;
+use warplib::warp_memory::WarpMemoryTrait;
+use warplib::warp_memory::WarpMemoryImpl;
+
+// ==================== Basic Functions ====================
+
+#[test]
+fn test_alloc(){
+    let mut warp_memory = WarpMemoryTrait::initialize();
+    assert(warp_memory.pointer == 0, 'Pointer should be 0');
+
+    warp_memory.unsafe_alloc(5);
+    assert(warp_memory.pointer == 5, 'Pointer should be 5');
+
+    warp_memory.alloc(10);
+    assert(warp_memory.pointer == 15, 'Pointer should be 15');
+}
+
+#[test]
+#[should_panic]
+fn test_alloc_overflow(){
+    let mut warp_memory = WarpMemoryTrait::initialize();
+
+    warp_memory.alloc(10);
+
+    // this should crash due to pointer overflow
+    warp_memory.alloc(-1);
+
+}
+
+#[test]
+fn test_unsafe_access() {
+    let mut warp_memory = WarpMemoryTrait::initialize();
+    warp_memory.unsafe_write(3, 5); 
+    
+    let read_from_unallocated = warp_memory.unsafe_read(3);
+    assert(read_from_unallocated == 5, 'Invalid read value');
+    assert(warp_memory.pointer == 0, 'Pointer should be 0');
+}
+
+#[test] 
+#[should_panic]
+fn test_unallocated_write() {
+    let mut warp_memory = WarpMemoryTrait::initialize();
+    warp_memory.write(3, 5); 
+}
+
+#[test] 
+#[should_panic]
+fn test_unallocated_read() {
+    let mut warp_memory = WarpMemoryTrait::initialize();
+    warp_memory.read(3); 
+}
```

Comment:
I suggest rearranging functions to group `#[should_panic]` together. It's hard to track whether the test should throw exception or not

Reply:
I think every rearregenment has it's pros and cons. The intent here was to arrange according to functionality since in my opinion is more natural and helpful. Indeed with this organization finding panicables test it would take more time, but in my experience I don't know many situations where that would be necessary.

Reply:
I agree, what I meant that you could rearrange panicables within the specific functionality to be near each other. Eg. it could be in the order: test_access,  test_unsafe_access, test_alloc, test_alloc_overflow, test_unallocated_write,  test_unallocated_read. 

I didn't want to group them  for the purpose of searching them, it's rather the fact it's easy to miss the `#[should_panic]` annotation as I've been reading through it.  If you prefer any other solution to make it more visible it's fine for me. E.g. names of tests could be more verbose like: `test_write_to_unallocated_space_should_fail`

Reply:
I think `test` is a good prefix and `should_panic` is a good suffix even if they are both described by attributes. I left the first and added the second

Reply:
Ok, fine by me